config/routes.rb

```diff
@@ -3,24 +3,23 @@
     registrations: 'pharmacies/registrations',
     sessions: 'pharmacies/sessions'
   }
+  devise_scope :pharmacy do
+    get 'pharmacies/show/:id' => 'pharmacies/registrations#show', as: :pharmacies_detail
+    get 'pharmacies/search' => 'pharmacies/registrations#search'
+    delete 'pharmacies/:id' => 'pharmacies/registrations#destroy'
+  end
   devise_for :users, :controllers => {
     registrations: 'users/registrations',
     sessions: 'users/sessions'
   }
-  devise_scope :pharmacy do
-    get 'pharmacies/show' => 'pharmacies/registrations#show'
-    get 'pharmacies/search' => 'pharmacies/registrations#search'
-    delete 'pharmacies/:id' => 'users/registrations#destroy'
-  end
   devise_scope :user do
     get 'users/show' => 'users/registrations#show'
     delete 'users/:id' => 'users/registrations#destroy'
   end
-  get '/pharmacies/show/:id', to: 'pharmacies/registrations#show'
   get '/opinions', to: 'homes#opinion'
   get '/histories', to: 'homes#histroy'
-  get '/conditions', to: 'conditions#index'
+  get '/conditions/index', to: 'conditions#index'
```

Comment:
一般的に `index`  はルーティングのところには記述しないので、元のままのほうが良かったかもです！
```suggestion
  get '/conditions', to: 'conditions#index'
```